### PR TITLE
Bump master to 4.3 branch

### PR DESCRIPTION
**Pull request by s-urbaniak**
This bumps `master` to be in sync with the current changes in `master-4.3`.
Once merged, we can delete `master-4.3` for good.

/cc @LiliC @paulfantom @brancz

### CODE REVIEW

**Comment by LiliC** [comment]
/lgtm

🤷‍♀

**Comment by LiliC** [comment]
Ah I thought it was tide job that was pending... nvm

**Comment by LiliC** [comment]
That looks like our failure actually.

```
-- FAIL: TestPodMetricsPresence (300.52s)
    prometheusadapter_test.go:155: number of running pods (138) doesn't match number of pods reporting metrics (0)
=== RUN   TestAggregatedMetricPermissions
```

**Comment by LiliC** [comment]
Same error again 🤔 

/hold

to hoperfully stop the tests being retriggered.

**Comment by s-urbaniak** [comment]
@LiliC @paulfantom please re-lgtm

File: jsonnet/jsonnetfile.json [hunk 1]
```diff
@@ -10,6 +10,16 @@
             },
             "version": "master"
         },
+        {
+            "name": "kube-thanos",
+            "source": {
+                "git": {
+                    "remote": "https://github.com/metalmatze/kube-thanos",
```
Comment (LiliC) [inline jsonnet/jsonnetfile.json]: Didn't this move? cc @metalmatze

Comment (LiliC) [reply jsonnet/jsonnetfile.json]: Yes seems like it did https://github.com/thanos-io/kube-thanos @s-urbaniak can you change this as well.

Comment (s-urbaniak) [reply jsonnet/jsonnetfile.json]: addressed, ptal :+1:

Comment (s-urbaniak) [reply jsonnet/jsonnetfile.json]: oh just a second, i think i bumped the wrong file.

**Pull request merged**